game.sy

```diff
@@ -233,22 +239,12 @@ init :: fn {
     push' bodies, puck
 
     d := 0.8
-    push' walls, collision.Wall {
-        normal: (-1.0, 0.0),
-        pos: (d, 0.0),
-    }
-    push' walls, collision.Wall {
-        normal: (1.0, 0.0),
-        pos: (-d, 0.0),
+    arena.walls -> for_each' fn wall: collision.Wall {
+        push' walls, wall
     }
 
-    push' corners, collision.Circle {
-        rad: d,
-        pos: (0.0, 1.0 - d),
-    }
-    push' corners, collision.Circle {
-        rad: d,
-        pos: (0.0, -1.0 + d),
+    arena.corners -> for_each' fn corner: collision.Circle {
```

Comment:
Can't you just overwrite the lists? Your already copying the elements by reference.